test/parallel/test-worker-max-count.js

```diff
@@ -0,0 +1,46 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const { spawn } = require('child_process');
+const { Worker } = require('worker_threads');
+
+if (process.argv[2] === 'child') {
+
+
+  common.expectWarning({
+    Warning: [
+      ['Too many active worker threads (2). Performance may be degraded. ' +
+       '(--max-worker-threads=1)'],
+      ['Too many active worker threads (2). Performance may be degraded. ' +
+       '(--max-worker-threads=1)']
+    ] });
+
+  const expr = 'setTimeout(() => {}, 10)';
```

Comment:
If the timeout is here to prevent one Worker thread from exiting before all workers are finished, I think 10 ms is going to lead to flakiness.

It’s probably better to first wait for `'online'` events from all Workers, then terminate them as a group. (`setInterval()` is a good way to keep them alive indefinitely, which should work here.)